File: csharp/Payloads/BookingDatesPayload.cs
using System;

namespace Payloads
{
    public class BookingDatesPayload
    {
        public  DateTime checkin { get; private set; }
        public  DateTime checkout { get; private set; }

        public BookingDatesPayload(DateTime checkin, DateTime checkout)
        {
            this.checkin = checkin;
            this.checkout = checkout;
        }
    }
}

File: java/springmvc/src/main/java/com/restfulbooker/api/payloads/request/BookingDatesPayload.java
package com.restfulbooker.api.payloads.request;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Date;

public class BookingDatesPayload {

    @JsonProperty
    private Date checkin;
    @JsonProperty
    private Date checkout;

    public BookingDatesPayload(Date checkin, Date checkout){
        this.checkin = checkin;
        this.checkout = checkout;
    }

    public Date getCheckin() {
        return checkin;
    }

    public Date getCheckout() {
        return checkout;
    }
}
